### PR TITLE
[server] Prevent avatarUrl update WEB-403

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-fix-avac0719b5426</li>
	<li><b>🔗 URL</b> - <a href="https://mp-fix-avac0719b5426.preview.gitpod-dev.com/workspaces" target="_blank">mp-fix-avac0719b5426.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>
 
/hold

### CODE REVIEW

File: components/server/src/container-module.ts [hunk 1]
```diff
@@ -206,8 +205,6 @@ export const productionContainerModule = new ContainerModule((bind, unbind, isBo
     bind(GitTokenScopeGuesser).toSelf().inSingletonScope();
     bind(GitTokenValidator).toSelf().inSingletonScope();
 
-    bind(BlockedUserService).to(BlockedUserService).inSingletonScope();
```
Comment (easyCZ) [inline components/server/src/container-module.ts:209]: Drive-by merges this service with the User-service. This was done to make stubbing out the UserService easier for tests

File: components/server/src/user/user-service.ts [hunk 2]
```diff
@@ -410,7 +406,7 @@ export class UserService {
             throw new ResponseError(ErrorCodes.NOT_FOUND, "User does not exist.");
         }
 
-        const allowedFields: (keyof User)[] = ["avatarUrl", "fullName", "additionalData"];
+        const allowedFields: (keyof User)[] = ["fullName", "additionalData"];
```
Comment (easyCZ) [inline components/server/src/user/user-service.ts:409]: this is the primary change

**Review by geropl — APPROVED** [review]
:heavy_check_mark:
